extensions/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/ResteasyStandaloneBuildStep.java

```diff
@@ -170,31 +98,23 @@ public void boot(ShutdownContextBuildItem shutdown,
         }
         feature.produce(new FeatureBuildItem(Feature.RESTEASY));
 
-        boolean isDefaultOrNullDeploymentPath = standalone.deploymentRootPath == null
-                || standalone.deploymentRootPath.equals("/");
-        if (!isDefaultOrNullDeploymentPath) {
-            // We need to register a special handler for non-default deployment path (specified as application path or resteasyConfig.path)
-            Handler<RoutingContext> handler = recorder.vertxRequestHandler(vertx.getVertx(), beanContainer.getValue(),
-                    executorBuildItem.getExecutorProxy(), httpConfiguration);
-            // Exact match for resources matched to the root path
-            routes.produce(new RouteBuildItem(standalone.deploymentRootPath, handler, false));
-            String matchPath = standalone.deploymentRootPath;
-            if (matchPath.endsWith("/")) {
-                matchPath += "*";
-            } else {
-                matchPath += "/*";
-            }
-            // Match paths that begin with the deployment path
-            routes.produce(new RouteBuildItem(matchPath, handler, false));
+        // Handler used for both the default and non-default deployment path (specified as application path or resteasyConfig.path)
+        // Routes use the order VertxHttpRecorder.DEFAULT_ROUTE_ORDER + 1 to ensure the default route is called before the resteasy one
+        Handler<RoutingContext> handler = recorder.vertxRequestHandler(vertx.getVertx(), beanContainer.getValue(),
+                executorBuildItem.getExecutorProxy(), httpConfiguration);
+        // Exact match for resources matched to the root path
+        routes.produce(new RouteBuildItem(
+                new BasicRoute(standalone.deploymentRootPath, VertxHttpRecorder.DEFAULT_ROUTE_ORDER + 1), handler));
```

Comment:
The order value of `VertxHttpRecorder.DEFAULT_ROUTE_ORDER + 1` means that resteasy route (resource methods) are called after the default route. Unlike custom routes and `@Route` methods which are called before the default route. The problem with resteasy route is that it returns 404 for unknown resource and does not call `RoutingContext.next()`. And since we handle the static resources in the default route it would always return 404 for a request to a static route. @cescoffier @stuartwdouglas Do you think this approach is OK? If yes, I'm going to merge this PR...